extensions/amp-viz-vega/0.1/amp-viz-vega.js

```diff
@@ -0,0 +1,161 @@
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {dev, user} from '../../../src/log';
+import {vsyncFor} from '../../../src/vsync';
+
+/** @const */
+const EXPERIMENT = 'amp-viz-vega';
+
+export class AmpVizVega extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    user.assert(isExperimentOn(this.win, EXPERIMENT),
+        `Experiment ${EXPERIMENT} disabled`);
+
+    /** @private {?JSONType} */
+    this.data_ = null;
+
+    /** @private {?string} */
+    this.inlineData_ = this.getInlineData_();
+
+    /** @private {?string} */
+    this.src_ = this.element.getAttribute('src');
+
+    user.assert(this.inlineData_ || this.src_,
+        '%s: neither `src` attribute nor a ' +
+        'valid <script type="application/json"> child was found for Vega data.',
+        this.getName_());
+
+    user.assert(!(this.inlineData_ && this.src_),
+        '%s: both `src` attribute and a valid ' +
+        '<script type="application/json"> child were found for Vega data. ' +
+        'Only one way of specifying the data is allowed.',
+        this.getName_());
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.initialize_();
+    return this.loadData_().then(() => this.renderGraph_());
+  }
+
+  /**
+   * Create the vega container.
+   * Called lazily in the first `#layoutCallback`.
+   */
+  initialize_() {
+    if (this.container_) {
+      return;
+    }
+
+    /** @private {?Element} */
+    this.container_ = this.element.ownerDocument.createElement('div');
+
+    this.applyFillContent(this.container_, true);
+    this.element.appendChild(this.container_);
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  loadData_() {
+    if (this.data_) {
+      return Promise.resolve();
+    }
+
+    // Validation in buildCallback should ensure one and only one of
+    // src_/inlineData_ is ever set.
+    dev.assert(!this.src_ != !this.inlineData_);
+
+    if (this.inlineData_) {
+      this.data_ = tryParseJson(this.inlineData_, err => {
+        user.assert(!err, 'data could not be ' +
+            'parsed. Is it in a valid JSON format?: %s', err);
+      });
+    } else {
+      // TODO(aghassemi): Fetch and validate the data file.
+      this.data_ = {
+        'url': this.src_,
+      };
+    }
+
+    return Promise.resolve();
+  }
+
+  /**
+   * @return {?string|undefined}
+   * @private
+   */
+  getInlineData_() {
+    let inlineConfig;
+
+    const scripts = childElementsByTag(this.element, 'SCRIPT');
+    if (scripts.length == 1) {
+      const child = scripts[0];
+      if (isJsonScriptTag(child)) {
+        inlineConfig = child.textContent;
+      } else {
+        user.error(this.getName_(), 'data should ' +
+            'be put in a <script type="application/json"> tag.');
+      }
+    } else if (scripts.length > 1) {
+      user.error(this.getName_(), 'more than one' +
```

Comment:
This should just be a `user.assert(scripts.length <= 1, ...)` so that the element throws an error during build.